### ISSUE TITLE
Cannot upgrade ExtAuth plugin to 1.7.2

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#4698]
**Describe the bug**
When customers recompile their extauth plugin to 1.7.2 
AND when this plugin fetches a secret using pkg `k8s.io/client-go v0.19.6`
Then, the plugin throws panic errors
Compile + Build are OK

**Expected behavior**
Stability over upgrades

**Additional context**
- Go 1.16
- Gloo Edge version 1.7.2
- Kubernetes version 1.17.12 and v1.18.9-eks

**Comment by kdorosh** [comment]
```
cat << EOF > plugin-values.yaml
global:
  extensions:
    extAuth:
      plugins:
        my-plugin:
          image:
            repository: gloo-ext-auth-plugins
            registry: quay.io/solo-io
            pullPolicy: IfNotPresent
            tag: 1.7.2
EOF
```

**Comment by kdorosh** [comment]
we were hitting https://github.com/golang/go/issues/45030 , the latest 1.7 uses golang 1.16.2 which has the bug, rather than 1.16.3 which has the fix.

The latest 1.8 betas are on 1.16.3, so we are upgrading and backporting those changes to 1.7.

Until the 1.7 release is out with the fix for the golang bug, users can try 1.8 beta releases which should work
